From e1cf07370ead445d5add403d9ae99a473c37108d Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Sun, 5 Apr 2020 15:38:32 -0700
Subject: [PATCH 07/14] tools/cmake: update to 3.17.0

Remove libressl patches; they are no longer needed as LibreSSl adde
support

Replace qt tests patch with one that disables all of them.

Refresh remaining one.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
---
 tools/cmake/Makefile                          |  6 ++--
 .../cmake/patches/100-disable_qt_tests.patch  | 34 -------------------
 tools/cmake/patches/100-no-testing.patch      | 33 ++++++++++++++++++
 .../110-libarchive-fix-libressl-compat.patch  | 11 ------
 .../130-bootstrap_parallel_make_flag.patch    |  2 +-
 .../patches/140-libarchive-fix-libressl.patch | 25 --------------
 6 files changed, 37 insertions(+), 74 deletions(-)
 delete mode 100644 tools/cmake/patches/100-disable_qt_tests.patch
 create mode 100644 tools/cmake/patches/100-no-testing.patch
 delete mode 100644 tools/cmake/patches/110-libarchive-fix-libressl-compat.patch
 delete mode 100644 tools/cmake/patches/140-libarchive-fix-libressl.patch

diff --git a/tools/cmake/Makefile b/tools/cmake/Makefile
index 336db5dd1f..ab3d225aa6 100644
--- a/tools/cmake/Makefile
+++ b/tools/cmake/Makefile
@@ -7,13 +7,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=cmake
-PKG_VERSION:=3.16.5
+PKG_VERSION:=3.17.0
 PKG_CPE_ID:=cpe:/a:kitware:cmake
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://github.com/Kitware/CMake/releases/download/v$(PKG_VERSION)/ \
-		https://cmake.org/files/v3.16/
-PKG_HASH:=5f760b50b8ecc9c0c37135fae5fbf00a2fef617059aa9d61c1bb91653e5a8bfc
+		https://cmake.org/files/v3.17/
+PKG_HASH:=b74c05b55115eacc4fa2b77a814981dbda05cdc95a53e279fe16b7b272f00847
 
 HOST_BUILD_PARALLEL:=1
 HOST_CONFIGURE_PARALLEL:=1
diff --git a/tools/cmake/patches/100-disable_qt_tests.patch b/tools/cmake/patches/100-disable_qt_tests.patch
deleted file mode 100644
index 9c0957a390..0000000000
--- a/tools/cmake/patches/100-disable_qt_tests.patch
+++ /dev/null
@@ -1,34 +0,0 @@
---- a/Tests/RunCMake/CMakeLists.txt
-+++ b/Tests/RunCMake/CMakeLists.txt
-@@ -375,15 +375,6 @@ else()
-   message(WARNING "Could not find or build ctresalloc")
- endif()
- 
--find_package(Qt4 QUIET)
--find_package(Qt5Core QUIET)
--if (QT4_FOUND AND Qt5Core_FOUND AND NOT Qt5Core_VERSION VERSION_LESS 5.1.0)
--  add_RunCMake_test(IncompatibleQt)
--endif()
--if (QT4_FOUND)
--  add_RunCMake_test(ObsoleteQtMacros -DQT_QMAKE_EXECUTABLE:FILEPATH=${QT_QMAKE_EXECUTABLE})
--endif()
--
- find_package(PkgConfig QUIET)
- if(PKG_CONFIG_FOUND)
-   add_RunCMake_test(FindPkgConfig)
---- a/Tests/CMakeLists.txt
-+++ b/Tests/CMakeLists.txt
-@@ -489,13 +489,6 @@ if(BUILD_TESTING)
- 
-   list(APPEND TEST_BUILD_DIRS ${CMake_TEST_INSTALL_PREFIX})
- 
--  if(NOT DEFINED CMake_TEST_Qt4)
--    set(CMake_TEST_Qt4 1)
--  endif()
--  if(CMake_TEST_Qt4 AND NOT QT4_FOUND)
--    find_package(Qt4 QUIET)
--  endif()
--
-   if(CMake_TEST_Qt4 AND QT4_FOUND)
-     # test whether the Qt4 which has been found works, on some machines
-     # which run nightly builds there were errors like "wrong file format"
diff --git a/tools/cmake/patches/100-no-testing.patch b/tools/cmake/patches/100-no-testing.patch
new file mode 100644
index 0000000000..501302b835
--- /dev/null
+++ b/tools/cmake/patches/100-no-testing.patch
@@ -0,0 +1,33 @@
+--- a/Modules/CTest.cmake
++++ b/Modules/CTest.cmake
+@@ -47,7 +47,7 @@ the :variable:`CTEST_USE_LAUNCHERS` variable::
+ in the ``CTestConfig.cmake`` file.
+ #]=======================================================================]
+ 
+-option(BUILD_TESTING "Build the testing tree." ON)
++option(BUILD_TESTING "Build the testing tree." OFF)
+ 
+ # function to turn generator name into a version string
+ # like vs9 or vs10
+--- a/Modules/Dart.cmake
++++ b/Modules/Dart.cmake
+@@ -33,7 +33,7 @@ whether testing support should be enabled.  The default is ON.
+ #
+ #
+ 
+-option(BUILD_TESTING "Build the testing tree." ON)
++option(BUILD_TESTING "Build the testing tree." OFF)
+ 
+ if(BUILD_TESTING)
+   find_package(Dart QUIET)
+--- a/Tests/Contracts/VTK/Dashboard.cmake.in
++++ b/Tests/Contracts/VTK/Dashboard.cmake.in
+@@ -25,7 +25,7 @@ ctest_empty_binary_directory(${CTEST_BINARY_DIRECTORY})
+ 
+ file(WRITE "${CTEST_BINARY_DIRECTORY}/CMakeCache.txt" "
+   BUILD_EXAMPLES:BOOL=ON
+-  BUILD_TESTING:BOOL=ON
++  BUILD_TESTING:BOOL=OFF
+   VTK_WRAP_PYTHON:BOOL=ON
+   ExternalData_OBJECT_STORES:FILEPATH=@base_dir@/ExternalData
+ ")
diff --git a/tools/cmake/patches/110-libarchive-fix-libressl-compat.patch b/tools/cmake/patches/110-libarchive-fix-libressl-compat.patch
deleted file mode 100644
index ba6565b6af..0000000000
--- a/tools/cmake/patches/110-libarchive-fix-libressl-compat.patch
+++ /dev/null
@@ -1,11 +0,0 @@
---- a/Utilities/cmlibarchive/libarchive/archive_openssl_evp_private.h
-+++ b/Utilities/cmlibarchive/libarchive/archive_openssl_evp_private.h
-@@ -28,7 +28,7 @@
- #include <openssl/evp.h>
- #include <openssl/opensslv.h>
- 
--#if OPENSSL_VERSION_NUMBER < 0x10100000L
-+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
- #include <stdlib.h> /* malloc, free */
- #include <string.h> /* memset */
- static inline EVP_MD_CTX *EVP_MD_CTX_new(void)
diff --git a/tools/cmake/patches/130-bootstrap_parallel_make_flag.patch b/tools/cmake/patches/130-bootstrap_parallel_make_flag.patch
index 9fcbf63f98..7b6198b14b 100644
--- a/tools/cmake/patches/130-bootstrap_parallel_make_flag.patch
+++ b/tools/cmake/patches/130-bootstrap_parallel_make_flag.patch
@@ -1,6 +1,6 @@
 --- a/bootstrap
 +++ b/bootstrap
-@@ -1248,7 +1248,10 @@ int main(){ printf("1%c", (char)0x0a); r
+@@ -1283,7 +1283,10 @@ int main(){ printf("1%c", (char)0x0a); r
  ' > "test.c"
  cmake_original_make_flags="${cmake_make_flags}"
  if [ "x${cmake_parallel_make}" != "x" ]; then
diff --git a/tools/cmake/patches/140-libarchive-fix-libressl.patch b/tools/cmake/patches/140-libarchive-fix-libressl.patch
deleted file mode 100644
index ddddc5e836..0000000000
--- a/tools/cmake/patches/140-libarchive-fix-libressl.patch
+++ /dev/null
@@ -1,25 +0,0 @@
-From 5da00ad75b09e262774ec3675bbe4d5a4502a852 Mon Sep 17 00:00:00 2001
-From: Bernard Spil <brnrd@FreeBSD.org>
-Date: Sun, 1 Apr 2018 23:01:44 +0200
-Subject: [PATCH] fix build with LibreSSL 2.7
-
-LibreSSL 2.7 adds OpenSSL 1.1 API leading to conflicts on method names
-
-See also: https://bugs.freebsd.org/226853
-Signed-off-by: Bernard Spil <brnrd@FreeBSD.org>
----
- libarchive/archive_openssl_hmac_private.h | 3 ++-
- 1 file changed, 2 insertions(+), 1 deletion(-)
-
---- a/Utilities/cmlibarchive/libarchive/archive_openssl_evp_private.h
-+++ b/Utilities/cmlibarchive/libarchive/archive_openssl_evp_private.h
-@@ -28,7 +28,8 @@
- #include <openssl/evp.h>
- #include <openssl/opensslv.h>
- 
--#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
-+#if OPENSSL_VERSION_NUMBER < 0x10100000L || \
-+	(defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x20700000L)
- #include <stdlib.h> /* malloc, free */
- #include <string.h> /* memset */
- static inline EVP_MD_CTX *EVP_MD_CTX_new(void)
-- 
2.34.1

